flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/Buffer.java

```diff
@@ -227,4 +224,58 @@
 	 * Tags the buffer as compressed or uncompressed.
 	 */
 	void setCompressed(boolean isCompressed);
+
+	/**
+	 * Gets the type of data this buffer represents.
+	 */
+	DataType getDataType();
+
+	/**
+	 * Sets the type of data this buffer represents.
+	 */
+	void setDataType(DataType dataType);
+
+	/**
+	 * Used to identify the type of data contained in {@link Buffer}.
+	 */
+	enum DataType {
+		/**
+		 * DATA_BUFFER indicates that this buffer represents a non-event data buffer.
+		 */
+		DATA_BUFFER(true),
+
+		/**
+		 * EXACTLY_ONCE_CHECKPOINT_BARRIER indicates that this buffer represents a
+		 * serialized checkpoint barrier of exactly once checkpoint mode.
+		 */
+		EXACTLY_ONCE_CHECKPOINT_BARRIER(false),
+
+		/**
+		 * UNKNOWN_EVENT indicates this buffer represents serialized data of other
+		 * unknown event.
+		 */
+		OTHER_EVENT(false);
```

Comment:
Some general usages also reference this, so I prefer to defining `EVENT_BUFFER` instead for general purpose. The semantic here is also covering the above `EXACTLY_ONCE_CHECKPOINT_BARRIER`. If we defined as `OTHER_EVENT`, it seems exclude the `EXACTLY_ONCE_CHECKPOINT_BARRIER`.